contrib/oauth2/mutt_oauth2.py

```diff
@@ -149,13 +149,16 @@ def writetokenfile():
     if not args.authorize:
         sys.exit('You must run script with "--authorize" at least once.')
     print('Available app and endpoint registrations:', *registrations)
-    token['registration'] = input('OAuth2 registration: ')
-    token['authflow'] = input('Preferred OAuth2 flow ("authcode" or "localhostauthcode" '
-                              'or "devicecode"): ')
+    token['registration'] = args.provider or input('OAuth2 registration: ')
+    token['authflow'] = args.authflow or input(
+        'Preferred OAuth2 flow ("authcode" or "localhostauthcode" or "devicecode"): '
+    )
     token['email'] = input('Account e-mail address: ')
```

Comment:
If we add an email cli arg in addition to the client_id and client_secret input() below, then auth can be done completely non-interactively, which would be nice.